scipy/stats/tests/test_stats.py

```diff
@@ -2245,21 +2245,32 @@ def test_mode_nan(self):
     ])
     def test_smallest_equal(self, data):
         result = stats.mode(data, nan_policy='omit')
-        assert_equal(result[0][0], 1)
+        assert_equal(result[0], 1)
 
     def test_obj_arrays_ndim(self):
         # regression test for gh-9645: `mode` fails for object arrays w/ndim > 1
         data = [['Oxidation'], ['Oxidation'], ['Polymerization'], ['Reduction']]
         ar = np.array(data, dtype=object)
         m = stats.mode(ar, axis=0)
-        assert np.all(m.mode == 'Oxidation') and m.mode.shape == (1, 1)
-        assert np.all(m.count == 2) and m.count.shape == (1, 1)
+        assert np.all(m.mode == 'Oxidation') and m.mode.shape == (1,)
+        assert np.all(m.count == 2) and m.count.shape == (1,)
 
         data1 = data + [[np.nan]]
         ar1 = np.array(data1, dtype=object)
         m = stats.mode(ar1, axis=0)
-        assert np.all(m.mode == 'Oxidation') and m.mode.shape == (1, 1)
-        assert np.all(m.count == 2) and m.count.shape == (1, 1)
+        assert np.all(m.mode == 'Oxidation') and m.mode.shape == (1,)
+        assert np.all(m.count == 2) and m.count.shape == (1,)
+
+    @pytest.mark.parametrize('axis', np.arange(-3, 3))
+    @pytest.mark.parametrize('dtype', [np.float64, 'object'])
+    def test_mode_shape_gh_9955(self, axis, dtype):
+        np.random.seed(984213899)
+        a = np.random.rand(3, 4, 5).astype(dtype)
```

Comment:
😉
```suggestion
        rng = np.random.default_rng(984213899)
        a = rng.random((3, 4, 5)).astype(dtype)
```

Reply:
Done.

Reply:
This could again be simplified using the ``dtype`` kwarg. i.e. ``rng.random((3, 4, 5), dtype=dtype)``

Reply:
Unfortunately not:
```
>>> a = rng.random(size=(3, 4, 5)).astype('O')  # no problem converting to object arrays

>>> a = rng.random(size=(3, 4, 5), dtype='O')  # problem
Traceback (most recent call last):

  Input In [23] in <module>
    a = rng.random(size=(3, 4, 5), dtype='O')

  File _generator.pyx:292 in numpy.random._generator.Generator.random

TypeError: Unsupported dtype dtype('O') for random
```